apps/experiments/models.py

```diff
@@ -1583,23 +1583,38 @@ def end(self, commit: bool = True, propagate: bool = True):
 
             enqueue_static_triggers.delay(self.id, StaticTriggerType.CONVERSATION_END)
 
-    def ad_hoc_bot_message(self, instruction_prompt: str, fail_silently=True, use_experiment: Experiment | None = None):
+    def ad_hoc_bot_message(
+        self,
+        name: str,
+        instruction_prompt: str,
+        fail_silently=True,
+        use_experiment: Experiment | None = None,
+        metadata: dict[str, Any] | None = None,
```

Comment:
How about we pass in something like `trace_info=TraceInfo(name=..., metadata=...)` instead of mixing trace specific params with the "normal" params?

Reply:
bd0c704e1ec4f94dd42ae5f894476183d628c863